File: Frontend/Nuka_POS/lib/models/currency.dart
class Currency {
  final int id;
  final String name;
  final String symbol;
  final String country;

  Currency({
    required this.id,
    required this.name,
    required this.symbol,
    required this.country,
  });

  factory Currency.fromJson(Map<String, dynamic> json) {
    return Currency(
      id: json['id'],
      name: json['name'],
      symbol: json['symbol'],
      country: json['country'],
    );
  }
}

File: Backend/nuka-pos/src/main/java/com/nuka/nuka_pos/api/currency/Currency.java
package com.nuka.nuka_pos.api.currency;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


/**
 * Currency entity represents a standard currency used within the system.
 * It defines a name and the symbol of the currency.
 */
@Entity
@Table(name = "currencies")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Currency {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String symbol;

    @Column(nullable = false)
    private String country;

}
